AsyncGuidance.md

```diff
@@ -593,8 +593,82 @@ public async Task<int> DoSomethingAsync()
 
 ## ConfigureAwait
 
-TBD
+`ConfigureAwait` changes whether the original thread context or thread scheduler is used when resuming a method after `await`.
 
+In practice, this means that code executing after `await` may or may not have access to global objects like `HttpContext`, or `SynchronizationContext.Current` may not be on the UI thread of a WPF application.
+
+`ConfigureAwait` takes a boolean value of `true` or `false`:
+* `ConfigureAwait(true)` will cause the async method to capture the context. This is the default behavior; `ConfigureAwait(true)` is rarely used.
+* `ConfigureAwait(false)` will cause the async method to not capture the context.
+
+Prefer `ConfigureAwait(false)` over the default behavior. Like `CancellationTokens`, this could be considered a cooperative effort.
+By not capturing the thread context or scheduler, less memory is used, and fewer CPU cycles are used when continuations runs.
+It also frees the application from having to run continuations through a `SynchronizationContext`.
+
+
+:bulb:**NOTE ASP.NET Core does not have a `SynchronizationContext` and does not have this problem.**
+
+:bulb:**NOTE Some advice states to use `ContinueAwait(false)` to avoid deadlocks. This is mistaken advice and does [not guarantee against deadlocks](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html).**
+
+❌ **BAD** This example uses a global object and throws a `NullReferenceException` during runtime.
```

Comment:
I think we should remove the ASP.NET examples (non core) as there's nothing else in this article that tries to explain that.